File: app/src/main/java/com/divercity/android/features/industry/onboarding/adapter/IndustryMultipleViewHolder.java
package com.divercity.android.features.industry.onboarding.adapter;

import android.support.v7.widget.AppCompatImageButton;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.divercity.android.R;
import com.divercity.android.data.entity.industry.IndustryResponse;

public class IndustryMultipleViewHolder extends RecyclerView.ViewHolder {

    private TextView txtTitle;
    private AppCompatImageButton imgBtn;
    private Listener listener;

    private IndustryMultipleViewHolder(View itemView, Listener listener) {
        super(itemView);
        txtTitle = itemView.findViewById(R.id.txt_title);
        imgBtn = itemView.findViewById(R.id.btn_public_private);
        this.listener = listener;
    }

    public void bindTo(IndustryResponse data) {
        txtTitle.setText(data.getAttributes().getName());

        imgBtn.setSelected(data.isSelected());
        imgBtn.setOnClickListener(view -> {
            imgBtn.setSelected(!data.isSelected());
            data.setSelected(imgBtn.isSelected());

            listener.onIndustryClick(data);
        });
    }

    public static IndustryMultipleViewHolder create(ViewGroup parent, Listener listener) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View view = layoutInflater.inflate(R.layout.item_btn_text, parent, false);
        return new IndustryMultipleViewHolder(view, listener);
    }

    public interface Listener {
        void onIndustryClick(IndustryResponse industry);
    }
}

File: app/src/main/java/com/divercity/android/features/jobs/jobs/search/searchfiltercompanyindustry/adapter/IndustryMultipleViewHolder.kt
package com.divercity.android.features.jobs.jobs.search.searchfiltercompanyindustry.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.divercity.android.R
import com.divercity.android.data.entity.industry.IndustryResponse
import kotlinx.android.synthetic.main.item_selection_text.view.*

class IndustryMultipleViewHolder private constructor(itemView: View, private val listener: Listener?) : RecyclerView.ViewHolder(itemView) {

    fun bindTo(selectStatus: Boolean, data: IndustryResponse?) {
        data?.let {
            itemView.apply {
                txt_title.text = data.attributes?.name

                btn_select_unselect.isSelected  = selectStatus

                setOnClickListener {
                    btn_select_unselect.isSelected = !btn_select_unselect.isSelected
                    listener?.onSelectUnselectIndustry(data, btn_select_unselect.isSelected)
                }
            }
        }
    }

    interface Listener {

        fun onSelectUnselectIndustry(industry: IndustryResponse, isSelected : Boolean)
    }

    companion object {

        fun create(parent: ViewGroup, listener: Listener?): IndustryMultipleViewHolder {
            val layoutInflater = LayoutInflater.from(parent.context)
            val view = layoutInflater.inflate(R.layout.item_selection_text, parent, false)
            return IndustryMultipleViewHolder(view, listener)
        }
    }
}
